core/src/main/java/io/fabric8/maven/core/handler/ContainerHandler.java

```diff
@@ -94,14 +87,19 @@ private String getImagePullPolicy(ResourceConfig config) {
     }
 
     private String getImageName(ImageConfiguration imageConfiguration) {
-        if (Strings.isNullOrBlank(imageConfiguration.getName())) {
-            return imageConfiguration.getName();
-        }
+        if (Strings.isNullOrBlank(imageConfiguration.getName()))
+            return null;
+        String configuredRegistry = EnvUtil.findRegistry(
+                Strings.isNullOrBlank(imageConfiguration.getName()) ? null : new ImageName(imageConfiguration.getName()).getRegistry(),
+                imageConfiguration.getRegistry(),
+                project.getProperties().getProperty("docker.pull.registry"),
```

Comment:
Not sure if its good idea to use `pull.registry ` as a general registry(for pull or push).

Reply:
Ideally we should also use system properties. You can use `Configs.getPropertyWithSystemAsFallback(project.getProperties(), "docker.pull.registry"`